File: Calcolatrice_C++/Model/figura.cpp
#include "figura.h"

int Figura::maxNomeLength=14; //francescominna

Figura::Figura(QString s)
{
    if(s.size() <= maxNomeLength)
        Nome=s;
    else
        throw std::invalid_argument("Nome troppo lungo!");
}

QString Figura::getNome() const
{
    return Nome;
}

void Figura::setNome(QString s)
{
    if(s.size() <= maxNomeLength)
        Nome=s;
    else
        throw std::invalid_argument("Nome troppo lungo!");
}

File: Calcolatrice_Java/src/Figura.java

public abstract class Figura{  
	private static final int maxLengthNome = 14;
	protected String name;
	
	public Figura(String name){
		setNome(name);
	}
	public Figura(){
		this("");
	}
	
	protected Figura(Figura f){
		this.name = f.getNome();
	}
	
	public String getNome(){
		return name;
	}
	public void setNome(String name){
		if(name.length()>14)
			throw new IllegalArgumentException("Nome troppo lungo!");
		this.name=name;
	}
	
	public abstract double getPerimetro();
	public abstract double getArea();
	
	public abstract Figura clone();
	
	@Override
	public String toString(){
		return name;
	}

	@Override
	public boolean equals(Object obj) {
		if(!(obj instanceof Figura)) 
			return false;

		if(this == obj) 
			return true;

		Figura f = (Figura)obj;
		return f.name == name;
	}
	
}